CDCgov/prime-reportstream
# Replace `History` reports API with `Delivery` API

This PR removes the use of the old report objects and API for the new Delivery API and response object. This includes:
- Delivery API hooks & object
- Fixes to API tooling

#### Test Steps:
1. Log into report stream as either a receiver or admin (if admin, mock as a receiver)
2. Navigate to the `Daily Data` page
3. Ensure the following three calls are present in your dev tools network tab: an org name like `md-phd`, `deliveries`, and `receivers`.
4. Click on the UUID of one to link to its details page.
5. Ensure the following calls are present: `delivery`, `facilities`

What's happening is on `/daily-data`, we call the org settings as a part of the page title, `deliveries` as the list of reports, and `receivers` fetches the list of org's receivers. The two latter calls are chained in a way that always guarantees we'll have an `org.service` variable to provide to the `deliveries` endpoint.

Loading `/details` pulls the UUID from the URL, plugs it into the `delivery` call, and fetches the report details. `facilities` is a separate call, and still uses the `rest-hooks` library.

`md-phd`, or your org of choice, will call again on `/details`, and that is due to the breadcrumbs using the same `useOrg` hook that the title on the previous page did. These calls, from what I can tell, are not the expensive part of this change...the only thing that's creating more network expense is the addition of the `receivers` call. I think we should consider moving this into an app-level state variable called `services` that fetches senders/receivers for your chosen org.

## Checklist

### Testing

### Process

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?
